tests/class-testcase.php

```diff
@@ -23,8 +31,73 @@ protected function setUp() {
 		parent::setUp();
 		Monkey\setUp();
 
+		// Mock roles to use across several tests.
+		$role1               = Mockery::mock( 'WP_Role' )->makePartial();
+		$role1->name         = 'Editor';
+		$role1->capabilities = [
+			'read'       => 'read',
+			'edit_books' => 'edit_books',
+			'edit_posts' => 'edit_posts',
+		];
+		$role1->allows(
+			[
+				'has_cap' => function( $cap ) {
+					return true;
+				},
+				'add_cap' => function( $cap ) {
+					return true;
+				},
+				'remove_cap' => function( $cap ) {
+				},
+			]
+		);
+
+		$role2               = Mockery::mock( 'WP_Role' )->makePartial();
+		$role2->name         = 'Administrator';
+		$role2->capabilities = [
+			'read'       => 'read',
+			'edit_books' => 'edit_books',
+			'edit_posts' => 'edit_posts',
+		];
+		$role2->allows(
+			[
+				'has_cap' => function( $cap ) {
+					return false;
+				},
+				'add_cap' => function( $cap ) {
+					return true;
+				},
+				'remove_cap' => function( $cap ) {
+				},
+			]
+		);
+
+		$role3               = Mockery::mock( 'WP_Role' )->makePartial();
+		$role3->name         = 'Subscriber';
+		$role3->capabilities = [];
+		$role3->allows(
+			[
+				'has_cap' => function( $cap ) {
+					return false;
+				},
+				'add_cap' => function( $cap ) {
+					return true;
+				},
+				'remove_cap' => function( $cap ) {
+				},
+			]
+		);
```

Comment:
Potentially remove duplicate code by saying `$role3 = $role1` and then configuring the different values?

Reply:
This would make it far less clear that they're objectively very different things. The only thing that would be 'cleaned up' by this approach, is the separate calls to Mockery, which would result in merely two lines being removed.

Reply:
Agreed to prefer this way for readability then 👍🏻